scripts/validationAndPreparation.sh

```diff
@@ -20,19 +22,18 @@ echo "files are $FILES"
 
 unset NEWEST_CONTRIBUTION
 
-#NEWEST_CONTRIBUTION="$(find_newer_contribution $DATE_OF_NEWEST_CONTRIBUTION)"
+#searching for newest contribution 
 echo "search for files newer than ${DATE_OF_NEWEST_CONTRIBUTION}"
 for f in $FILES
 do
 	if [[ !  $f == "challenges/" ]]; then
 		echo "Processing $f"
-		DATE=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $6}' | sed 's/[^0-9]*//g'`
-		echo "DATE is $DATE"
-		if [ $DATE -gt $DATE_OF_NEWEST_CONTRIBUTION ]; then
-			echo "found newer contribution"
-			DATE_OF_NEWEST_CONTRIBUTION=$DATE
+		FILE_CREATION_TIMESTAMP=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $6}' | sed 's/[^0-9]*//g'`
```

Comment:
You might be able to get date and filename with just one `ls` command. 

E.g. you could consider printing $6 and $7 concatenated, sorting that and then cutting off the date part. 
This might reduce the amount of long `lsftp` lines smaller and should also be more efficient.

The approach of doing one `cls */*` command in the first place did not work?

Reply:
your proposed solution down is working... Thanks